frontend/z3_axioms.py

```diff
@@ -388,6 +384,28 @@ def instance_axioms(called, args, result, types):
     return axioms
 
 
+def func_call(called, args, result, types):
+    if len(args) == 0:
+        return called == types.funcs[0](result)
+
+    subtype_axioms = []
+    z3_args = []
+    for i in range(len(args)):
+        z3_arg = getattr(types.type_sort, "func_{}_arg_{}".format(len(args), i + 1))(called)
+        z3_args.append(z3_arg)
+        arg = args[i]
+        subtype_axioms.append(And(
+            Implies(And(types.subtype(arg, types.num), types.subtype(z3_arg, types.num)),
```

Comment:
IF that is the case, then yes, I guess. This is what they do in MyPy: http://mypy.readthedocs.io/en/latest/duck_type_compatibility.html